### ISSUE TITLE
Fails for Generic types

### ISSUE THREAD

**Issue by donpatrice** [mkorpela/overrides#22]
The following fails:
```
from typing import Generic, TypeVar

from overrides import overrides

TObject = TypeVar("TObject", bound="Foo")


class Foo(Generic[TObject]):
    """Foo class."""
    def foo_method(self):
        """Foo method."""
        pass


class Bar(Foo["Bar"]):
    """Bar class."""
    @overrides
    def foo_method(self):
        return 5


converter = Bar()
print(converter.foo_method())
```
with
```
Traceback (most recent call last):
  File "bla.py", line 15, in <module>
    class Bar(Foo["Bar"]):
  File "bla.py", line 17, in Bar
    @overrides
  File "/Users/patrik/.local/share/virtualenvs/dbbsim-j0PjGvEt/lib/python3.7/site-packages/overrides/overrides.py", line 70, in overrides
    method.__name__)
AssertionError: No super class method found for "foo_method"
```

**Comment by mkorpela** [comment]
@donpatrice Thank you for the contribution!

**Comment by mkorpela** [comment]
Released in version 2.0
